crates/bevy_pbr/src/light_probe/generate.rs

```diff
@@ -0,0 +1,1084 @@
+
+                let tex_size = env_map_light.environment_map.size;
+                let wg_x = (tex_size.width / 64).max(1);
+                let wg_y = (tex_size.height / 64).max(1);
+                compute_pass.dispatch_workgroups(wg_x, wg_y, 6); // 6 faces
+            }
+
+            // Second pass - process mips 6-12
+            {
+                let mut compute_pass =
+                    render_context
+                        .command_encoder()
+                        .begin_compute_pass(&ComputePassDescriptor {
+                            label: Some("spd_second_pass"),
+                            timestamp_writes: None,
+                        });
+
+                compute_pass.set_pipeline(spd_second_pipeline);
+                compute_pass.set_bind_group(0, &bind_groups.spd, &[]);
+
+                let tex_size = env_map_light.environment_map.size;
+                let wg_x = (tex_size.width / 256).max(1);
+                let wg_y = (tex_size.height / 256).max(1);
+                compute_pass.dispatch_workgroups(wg_x, wg_y, 6);
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// Radiance map node for generating specular environment maps
+pub struct RadianceMapNode {
+    query: QueryState<(
+        Entity,
+        Read<GeneratorBindGroups>,
+        Read<RenderEnvironmentMap>,
+    )>,
+}
+
+impl FromWorld for RadianceMapNode {
+    fn from_world(world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(world),
+        }
+    }
+}
+
+impl Node for RadianceMapNode {
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
+        &self,
+        _graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let pipelines = world.resource::<GeneratorPipelines>();
+
+        let Some(radiance_pipeline) = pipeline_cache.get_compute_pipeline(pipelines.radiance)
+        else {
+            return Ok(());
+        };
+
+        for (_, bind_groups, env_map_light) in self.query.iter_manual(world) {
+            let mut compute_pass =
+                render_context
+                    .command_encoder()
+                    .begin_compute_pass(&ComputePassDescriptor {
+                        label: Some("radiance_map_pass"),
+                        timestamp_writes: None,
+                    });
+
+            compute_pass.set_pipeline(radiance_pipeline);
+
+            let base_size = env_map_light.specular_map.size.width;
+
+            // Process each mip level
+            for (mip, bind_group) in bind_groups.radiance.iter().enumerate() {
+                compute_pass.set_bind_group(0, bind_group, &[]);
+
+                // Calculate dispatch size based on mip level
+                let mip_size = base_size >> mip;
+                let workgroup_count = mip_size.max(8) / 8;
```

Comment:
I think you actually want div_ceil(8) here

Reply:
this and the other div_ceil comments are not actually an issue because there are asserts guaranteeing divisibility, so this is just about handling the zero case. feel free to mark resolved

Reply:
resolved by https://github.com/mate-h/bevy/pull/8